mono/utils/mono-merp.c

```diff
@@ -193,151 +195,49 @@ parse_exception_type (const char *signal)
-{
-	// The format is arch-specific
-#if defined(TARGET_AMD64)
-	g_string_append_printf (output, "%016llx\n", ptr);
-#elif defined(TARGET_X86)
-	g_string_append_printf (output, "%08x\n", ptr);
-#endif
-}
-
-static void
-print_memory_fraction (GString *output, size_t mem_bytes)
-{
-	if (!mem_bytes) {
-		g_string_append_printf (output, "\n");
-		return;
-	}
-
-	mem_bytes >>= 10; // convert from bytes to kb
-	float frac = (float) mem_bytes / 1024.0; // kb to mb
-
-	g_string_append_printf (output, "%f\n", frac);
-}
-
 static void
 mono_encode_merp (GString *output, MERPStruct *merp)
 {
-	// Format of integers seems to be 8 digits with padding
-	g_assert (merp->pidArg);
-	g_string_append_printf (output, "%.8x\n", merp->pidArg);
-
-	g_assert (merp->archArg);
-	append_merp_arch (output, merp->archArg);
-
-	g_string_append_printf (output, "%.8x\n", merp->capabilitiesArg);
-	print_pointer_param (output, merp->threadArg);
-	print_pointer_param (output, merp->timeArg);
-
-	append_merp_exctype (output, merp->exceptionArg);
-
-	g_assert (merp->bundleIDArg);
-	g_assert (merp->signatureArg);
-	g_assert (merp->versionArg);
-	g_string_append_printf (output, "%s\n%s\n%s\n", merp->bundleIDArg, merp->signatureArg, merp->versionArg);
+	// Provided by icall
+	g_string_append_printf (output, "ApplicationBundleID: %s\n", merp->bundleIDArg);
+	g_string_append_printf (output, "ApplicationVersion: %s\n", merp->versionArg);
 
-	print_string_or_blank (output, merp->devRegionArg);
-	print_string_or_blank (output, merp->uiLidArg);
+	g_string_append_printf (output, "ApplicationBitness: %s\n", get_merp_bitness (merp->archArg));
 
-	g_assert (merp->serviceNameArg);
-	g_string_append_printf (output, "%s\n", merp->serviceNameArg);
+	// Provided by icall
+	g_string_append_printf (output, "ApplicationName: %s\n", merp->serviceNameArg);
 
-	print_string_or_blank (output, merp->appLoggerName);
-	print_string_or_blank (output, merp->appLogSessionUUID);
+	// When embedded, sometimes this really doesn't make sense.
+	// FIXME: On OSX, could figure this out for just VS4Mac.
+	g_string_append_printf (output, "ApplicationPath: %s\n", merp->servicePathArg ? merp->servicePathArg : "missing");
 
-	g_string_append_printf (output, "%d\n", merp->isOfficeApplication ? 1 : 0);
+	// Provided by icall
+	g_string_append_printf (output, "BlameModuleName: %s\n", merp->moduleName);
+	g_string_append_printf (output, "BlameModuleVersion: %s\n", merp->moduleVersion);
+	g_string_append_printf (output, "BlameModuleOffset: 0x%x\n", merp->moduleOffset);
 
-	print_string_or_blank (output, merp->isObjCException ? "1" : NULL);
+	g_string_append_printf (output, "ExceptionType: %s\n", get_merp_exctype (merp->exceptionArg));
 
-	print_memory_fraction (output, merp->memRes);
-	print_memory_fraction (output, merp->memVirt);
+	g_string_append_printf (output, "StackChecksum: %x\n", merp->hashes.offset_free_hash);
+	g_string_append_printf (output, "StackHash: %x\n", merp->hashes.offset_rich_hash);
```

Comment:
can we either use `%p` or `0x%08x`/`0x%016x`?

also, what's the reasoning for the naming? we should have it consistent between the output and the names in the struct.